uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/Service/DrupalUserServiceTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use Civi\Test;
+use CRM_HRCore_Service_DrupalUserService as DrupalUserService;
+use CRM_HRCore_Service_DrupalRoleService as DrupalRoleService;
+
+/**
+ * @group headless
+ */
+class DrupalUserServiceTest extends \PHPUnit_Framework_TestCase implements HeadlessInterface, TransactionalInterface {
+
+  /**
+   * @var string
+   */
+  private $testEmail = 'foo@bar.com';
+
+  /**
+   * @var array
+   */
+  private $testContact;
+
+  public function setUpHeadless() {
+    return Test::headless()->installMe(__DIR__)->apply();
+  }
+
+  public function setUp() {
+    $this->cleanup();
+    $params = ['email' => $this->testEmail];
+    $this->testContact = CRM_HRCore_Test_Fabricator_Contact::fabricate($params);
+    $this->registerCurrentLoggedInContactInSession($this->testContact['id']);
+  }
+
+  public function tearDown() {
+    $this->cleanup();
+  }
+
+  public function testBasicUserCreate() {
+    $roleService = $this->prophesize(DrupalRoleService::class);
+    $drupalUserService = new DrupalUserService($roleService->reveal());
+    $user = $drupalUserService->createNew(
+      $this->testContact['id'],
+      $this->testEmail
```

Comment:
I realized it's actually pointless to pass 'email' into the fabricator since it's a separate entity :smile: 

I'll update it to not pass email in, but I'll still need the email as a property to clean up the contact afterwards.